nvflare/edge/assessors/general_num.py

```diff
@@ -0,0 +1,234 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import random
+import threading
+import time
+from typing import Optional
+
+from nvflare.apis.dxo import DXO
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+from nvflare.edge.aggregators.num_dxo import NumDXOAggregator
+from nvflare.edge.assessor import Assessment, Assessor
+from nvflare.edge.mud import BaseState, ModelUpdate, StateUpdateReply, StateUpdateReport
+
+
+class _ModelState:
+
+    def __init__(self, aggr: NumDXOAggregator):
+        self.aggregator = aggr
+        self.devices = {}
+        self.last_update_time = None
+
+    def accept(self, model_update: ModelUpdate, fl_ctx: FLContext):
+        self.last_update_time = time.time()
+        self.devices.update(model_update.devices)
+        return self.aggregator.accept(model_update.update, fl_ctx)
+
+
+class GeneralNumAssessor(Assessor):
```

Comment:
This class and GeneralModelUpdate assessor are very similar. Consider consolidating them. Build a base class that implements common functionalities. Define abstract methods for their differences.